backend/app/deployment_config.py

```
@@ -439,6 +452,14 @@ async def update_deployment_config_value(
         except ValueError:
             raise HTTPException(status_code=400, detail="RAG_TOP_K must be between 1 and 100")
 
+    if key in RATE_LIMIT_KEYS and value_to_save:
+        try:
+            rate_limit = int(value_to_save)
+            if rate_limit < 1:
+                raise ValueError()
+        except ValueError:
+            raise HTTPException(status_code=400, detail=f"{key} must be a positive integer")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)